app/uk/gov/hmrc/exports/mongock/changesets/CacheChangeLog.scala

```diff
@@ -120,27 +131,27 @@ class CacheChangeLog {
       val setUpdate = ImmutableMap.of("$set", ImmutableMap.of("locations.destinationCountry", codeElement))
       val setUpdateObj = new BasicDBObject(setUpdate)
       logger.info(s"Applying [4] changes to ${document.get("id")}")
-      db.getCollection(collection).findOneAndUpdate(objectToBeUpdated, setUpdateObj)
-      logger.info(s"Applied [4] changes to ${document.get("id")}")
+      getDeclarationsCollection(db).findOneAndUpdate(objectToBeUpdated, setUpdateObj)
     }
     logger.info("Applying 'CEDS-2247 Change destination country structure' db migrations... Done.")
   }
 
   @ChangeSet(order = "005", id = "CEDS-2247 Change routing countries structure", author = "Patryk Rudnicki")
   def changeRoutingCountriesStructure(db: MongoDatabase): Unit = {
+
     logger.info("Applying 'CEDS-2247 Change routing countries structure... ")
     val queryParams: ImmutableMap[String, AnyRef] =
       ImmutableMap.of("locations.routingCountries", ImmutableMap.of("$exists", true, "$ne", ""))
     val query: Document = new Document(queryParams)
 
-    val documents = db.getCollection(collection).find(new BasicDBObject(query)).toSeq
+    val documents = getDeclarationsCollection(db).find(new BasicDBObject(query))
```

Comment:
I'm worried that this approach was wrong. This load all declarations to the memory and that was one of the problems on staging (around half of the million declarations).

Paulo's approach from JavaChangeLog is using mongo to perform updates, without loading the all declarations in that way.

Reply:
Regarding loading eagerly/lazily this code behaves in exactly the same way as the Java code you mention. Except the difference that I am assigning the result to a value.
The `find` method returns `FindIterable` which (although not guaranteed in futurre versions) is taking documents lazily in batches (to limit amount of DB calls): https://stackoverflow.com/questions/39161042/is-finditerabledocument-load-all-documents

Reply:
That's a good point, but Java methods right now filter more specifically what need to be updated.

I'll leave it to @PauloPortugal he spent more time on those migrations and knows more about it :)